rundeck-authz/rundeck-authz-api/src/main/java/com/dtolabs/rundeck/core/authorization/RuleEvaluator.java

```diff
@@ -150,6 +150,24 @@ private static boolean matchesContexts(
                     return true;
                 }
             }
+
+            //rule urn match for user: urn against username
+            if (subject.getUsername() != null && rule.getUrn() != null && rule.getUrn().startsWith("user:")) {
+                if (("user:"+subject.getUsername()).equals(rule.getUrn())) {
+                    return true;
+                }
+            }
+            //rule urn match for role: urn against groups
+            if (subject.getGroups() != null && rule.getUrn() != null && rule.getUrn().startsWith("group:")) {
+                if (subject
+                        .getGroups()
+                        .stream()
+                        .map(a -> "group:" + a)
+                        .collect(Collectors.toSet())
+                        .contains(rule.getUrn())) {
+                    return true;
+                }
+            }
```

Comment:
should this code be added to notBy clause?

Reply:
good catch